core-test/src/test/scala/org/bitcoins/core/crypto/bip32/BIP32PathTest.scala

```diff
@@ -127,4 +128,67 @@ class BIP32PathTest extends BitcoinSUnitTest {
       assert(path == BIP32Path.fromString(toString))
     }
   }
+
+  it must "do path diffing" in {
+    {
+      val first = BIP32Path.fromString("m/44'/1'")
+      assert(first.diff(first).contains(BIP32Path.empty))
+    }
+
+    {
+      val first = BIP32Path.fromString("m/44'/0'/0'")
+      val second = BIP32Path.fromString("m/44'/0'/0'/0/2")
+      val expected = BIP32Path.fromString("m/0/2")
```

Comment:
So this seems weird to me, why do we keep the `m` prefix? Isn't this going to confuse people to believe that this is an _absolute_ path and not a _relative_ one?

Reply:
Same comment applies to the rest of these test vectors i believe

Reply:
That's just because all paths must begin with `m/at-least-one-level`. 

All paths are relative in HD schemes anyway, as it all depends on what your vantage point (your xpriv) and direction (your path) are.

Something we could add though (although I'd do it a later point), is a wrapper around an xpriv/xpub and a path, allowing us to go back and forth, having a notion of relativeness/absoluteness